PFW-1354 Fix slow button processing

gudnimg opened the pull request:
JIRA ticket: https://dev.prusa3d.com/browse/PFW-1354

The issue on the MK3S side:
1. There is an MMU error screen
2. Select Retry
3. The error screen is dismissed. MK3S sends Q0 MMU Query
4. MMU Query returns back the error state
5. At this point we process the button (a bit late)

We should process the button just after we dismiss the error screen and before sending another Query Q0 request.

gudnimg committed:
Fix MK3S buttons being processed too slowly

After a button is selected on the MK3S MMU error screen, we need to process it before asking for another MMU Query.

Because the MMU Query will make the Error screen return and overwrite the status screen

gudnimg commented:
This needs to be tested well. I don't see any issue with this from the MK3S perspective. But could there be an issue where the MMU buttons are pushed at random times?

My thought is when the MMU error screen is NOT present, then `CheckUserInput` should not do anything. But I'm not sure about the physical buttons on the MMU, are there times they should not do anything?

Here is an image showing the serial stream with the above fix, it looks good.
![image](https://user-images.githubusercontent.com/8218499/175769992-f9d11d78-d451-468b-b1a6-7213232cc498.png)

vintagepc commented:
Physical MMU buttons won't be affected unless during a toolchange when the "printer in charge" mode is enabled.
But you're right it has an impact on that if we want to use this mode for other things.

vintagepc commented on Firmware/mmu2.cpp:
```diff
@@ -638,27 +639,22 @@ StepStatus MMU2::LogicStep() {
     case ButtonPushed:
         lastButton = logic.Button();
         LogEchoEvent("MMU Button pushed");
-        CheckUserInput();
```
This one might need to be left alone otherwise the MMU button won't be registered until the next cycle, if at all.

gudnimg replied:
I've added this back

gudnimg committed:
Add back CheckUserInput for ButtonPushed event

vintagepc committed:
PFW-1351 Cooldown timeout (#33)

* PFW-1351 WIP

* Fix bug with cooldown timing due to bad bitfield.

* Use default safety timer value for timeout

gudnimg committed:
Fix PFW-1357 (#35)

lcd_encoder_diff would be set to 0 if lcd_update_enable(true)
is called outside ReportErrorHookMonitor
This would put enc_dif out of sync with lcd_encoder_diff and could cause false rotation triggering

gudnimg commented:
Rebased branch onto mk3-mmu

gudnimg commented:
There is currently a known issue with the MMU buttons, where the MMU reports the button to the MK3S multiple times. This seems to be unrelated to the changes in this PR. This PR mainly resolves the LCD glitches observed when using the MMU screen.

gudnimg commented:
@vintagepc Its very strange that I have not seen the issue this PR was supposed to solve again. The LCD behaved well this weekend 🤔 Smells funny, maybe we should wait with merging this one? What's your opinion?
